### ISSUE TITLE
PURPLE germline-only run

### ISSUE THREAD

**Issue by TuanTrain** [hartwigmedical/hmftools#291]
Hello,

I'm trying to run PURPLE in the germline-only mode, as is referenced in the documentation. There is no example to follow, and trying out different options such as `-germline_only` or `-germline` or `-reference_only` does not work.

Is this feature available and if so, how should it be run?

Also, in the minimum set of arguments, the `-ensembl_data_dir` is not mentioned, but in trying to run the latest version (3.3), there is an error given stating it is required. What should I put for that param when I don't have anything for it?

Thank you for the assistance!
Antuan

**Comment by charlesshale** [comment]
I've just updated the Purple read-me with examples for germline-only and tumor-only.

So to run Purple in germline-only mode, just omit the 'tumor' sampleId.

Yes the Ensembl data cache is required and I've also added that to a few of the example commands which were missing it.

thanks.

**Comment by TuanTrain** [comment]
Hi, I tried to run it based on the suggested usage, but I still get this error:

23:40:54 - [INFO ] - PURPLE version: 3.3
23:40:54 - [ERROR] - Missing the following parameters: tumor
23:40:54 - [INFO ] - Reference Sample: XYZ, Tumor Sample: 
23:40:54 - [INFO ] - Output Directory: /xyz/
23:40:54 - [ERROR] - initialisation error, exiting

It seems that the validation is not allowing the program to even begin.

Please let me know if there's a workaround or perhaps a quick fix. Thanks!

**Comment by charlesshale** [comment]
Could you use Purple v3.4:

https://github.com/hartwigmedical/hmftools/releases/tag/purple-v3.4

**Comment by TuanTrain** [comment]
The newer version fixed the previous issue, but this new one came up:

00:10:52 - [INFO ] - generating QC Stats
00:10:52 - [INFO ] - generating charts
00:10:52 - [ERROR] - failed processing sample(XYZ): java.lang.NullPointerException
java.lang.NullPointerException
	at com.hartwig.hmftools.purple.PurpleApplication.performFit(PurpleApplication.java:451)
	at com.hartwig.hmftools.purple.PurpleApplication.processSample(PurpleApplication.java:235)
	at com.hartwig.hmftools.purple.PurpleApplication.run(PurpleApplication.java:176)
	at com.hartwig.hmftools.purple.PurpleApplication.main(PurpleApplication.java:642)

Any ideas?

I was mainly looking to get the plots from Purple, so unfortunate that it is exactly where it failed.

**Comment by charlesshale** [comment]
Could you share the config you're using for Purple?

**Comment by TuanTrain** [comment]
Not sure what you mean by the config. I did install the R dependencies for the plots, and here are the relevant params I am inputting: 

java -jar purple_v3.4.1.jar \
   -gc_profile GC_profile.1000bp.38.cnp \
   -ref_genome_version 38 \
   -ref_genome Homo_sapiens_assembly38.fasta \
   -ensembl_data_dir Ensembl-Data-Cache/38

**Comment by charlesshale** [comment]
Ok thanks. In germline mode, Purple cannot produce any plots and it's intended that the 'circos' config is left out and that 'no_charts' is also configured. I'll change it to ignore those config options automatically when in germline-only mode.

Which plots were you interested in seeing for the germline variants?

**Comment by TuanTrain** [comment]
got it. I used another sample as a reference and ran in normal mode to get the plots. Was looking for the "xyz.input.png" plot, but for a small window (10k BP). It seems that the bins however are too coarse. Is there a way to rerun Amber/ Cobalt/ Purple with more fine grained bins, perhaps specifying only the necessary window? I found a way in Purple to change the circos config file, but it seems that the bins are set earlier.

Thanks for the continued help!
